tests/unit-tests/scene/test_basic_idle_hub.cpp

```diff
@@ -252,17 +252,121 @@ TEST_F(BasicIdleHub, observer_marked_idle_after_shorter_timeout_removed_and_poke
     auto const observer2 = std::make_shared<StrictMock<MockObserver>>();
     EXPECT_CALL(*observer1, active()).Times(AnyNumber());
     EXPECT_CALL(*observer2, active()).Times(AnyNumber());
-    hub.register_interest(observer1, executor, 5s);
-    hub.register_interest(observer2, executor, 10s);
+    hub->register_interest(observer1, executor, 5s);
+    hub->register_interest(observer2, executor, 10s);
     executor.execute();
     EXPECT_CALL(*observer1, idle());
     advance_by(6s);
     executor.execute();
-    hub.unregister_interest(*observer1);
-    hub.poke();
+    hub->unregister_interest(*observer1);
+    hub->poke();
     advance_by(9s);
     executor.execute();
     EXPECT_CALL(*observer2, idle());
     advance_by(2s);
     executor.execute();
 }
+
+TEST_F(BasicIdleHub, inhibit_idle_inhibits_until_wake_lock_released)
+{
+    auto const observer = std::make_shared<NiceMock<MockObserver>>();
+    hub->register_interest(observer, executor, 5s);
+    // Check that it goes idle
+    EXPECT_CALL(*observer, idle());
+    advance_by(6s);
+    executor.execute();
+    // Check that it stays awake if idle inhibited
+    hub->poke();
+    {
+        auto wake_lock = hub->inhibit_idle();
+        EXPECT_CALL(*observer, active());
+        advance_by(6s);
+        executor.execute();
+    }
+    // Check that it goes idle when wake_lock goes out of scope
+    EXPECT_CALL(*observer, idle());
```

Comment:
I think this expectation belongs to another test:

`when_a_wake_lock_is_released_idle_timeout_is_restarted`